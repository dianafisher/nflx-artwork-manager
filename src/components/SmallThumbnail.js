import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './SmallThumbnail.css';

class SmallThumbnail extends Component {

  _onThumnbailClicked = () => {
    this.props.onClick(this.props.data);
  }

  render() {
    const data = this.props.data;
    
    return (
      <div className='small-thumbnail'>
        <img
          src={data.thumbnailUrl}
          alt='small thumbnail'
          onClick={this._onThumnbailClicked}
        ></img>
      </div>
    )
  }
}

SmallThumbnail.propTypes = {
  onClick: PropTypes.func.isRequired,
  data: PropTypes.object.isRequired
};

export default SmallThumbnail;
