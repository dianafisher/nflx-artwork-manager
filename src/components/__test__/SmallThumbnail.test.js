import React from 'react';
import { shallow } from 'enzyme';
import SmallThumbnail from '../SmallThumbnail';

const data = {
  thumbnailUrl: 'http://via.placeholder.com/350x150'
};

describe('SmallThumbnail', () => {

  it('renders', () => {
    expect(shallow(
      <SmallThumbnail data={data} onClick={jest.fn()}/>
    )).toMatchSnapshot();
  });

  it('calls _onThumnbailClicked when clicked', () => {
    const onClicked = jest.fn();
    const wrapper = shallow(
      <SmallThumbnail data={data} onClick={onClicked}/>
    );

    wrapper.find('img').simulate('click');
    expect(onClicked).toBeCalled();
  });
});
