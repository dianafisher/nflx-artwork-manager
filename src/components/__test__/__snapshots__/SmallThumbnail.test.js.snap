// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallThumbnail renders 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="small-thumbnail"
>
    <img
        alt="small thumbnail"
        onClick={[Function]}
        src="http://via.placeholder.com/350x150"
    />
</div>,
  "nodes": Array [
    <div
      className="small-thumbnail"
>
      <img
            alt="small thumbnail"
            onClick={[Function]}
            src="http://via.placeholder.com/350x150"
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SmallThumbnail
        data={
                Object {
                        "thumbnailUrl": "http://via.placeholder.com/350x150",
                      }
        }
        onClick={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SmallThumbnail {
        "_onThumnbailClicked": [Function],
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "data": Object {
            "thumbnailUrl": "http://via.placeholder.com/350x150",
          },
          "onClick": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="small-thumbnail"
>
          <img
                    alt="small thumbnail"
                    onClick={[Function]}
                    src="http://via.placeholder.com/350x150"
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="small-thumbnail"
>
          <img
                    alt="small thumbnail"
                    onClick={[Function]}
                    src="http://via.placeholder.com/350x150"
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SmallThumbnail
    data={
        Object {
            "thumbnailUrl": "http://via.placeholder.com/350x150",
          }
    }
    onClick={[Function]}
/>,
}
`;
